File: LeetCode/122. 买卖股票的最佳时机 II.ts
function maxProfit(prices: number[]): number {
    let buyPoint: number = -1;
    let inCome: number = 0;
    for (let i = 0;i < prices.length;i ++) {
        if (buyPoint == -1 || prices[buyPoint] > prices[i]) {
            buyPoint = i;
        }
        if (buyPoint != -1 && prices[i] > prices[buyPoint]) {   
            inCome += prices[i] - prices[buyPoint];
            buyPoint = i;
        }
    }
    return inCome;
};

File: LeetCode/122. 买卖股票的最佳时机 II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        int buyPrice = 100001
        int earning = 0
        for  price in prices:
            if price < buyPrice:
                buyPrice = price
            else:
                earning += price-buyPrice
                buyPrice = price
        return earning